File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/ChartManager.java
package com.remotearthsolutions.expensetracker.utils;

import android.content.Context;
import com.remotearthsolutions.expensetracker.entities.ExpeneChartData;

import java.util.List;

public interface ChartManager {

    void initPierChart(int deviceDp, Utils.ScreenSize screenSize);

    void loadExpensePieChart(Context context, ChartManagerImpl.ChartView chartView, List<ExpeneChartData> data);

}

File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/ChartManager.kt
package com.remotearthsolutions.expensetracker.utils

import android.content.Context
import com.remotearthsolutions.expensetracker.entities.ExpeneChartData
import com.remotearthsolutions.expensetracker.utils.Utils.ScreenSize

interface ChartManager {
    fun initPierChart(deviceDp: Int, screenSize: ScreenSize?)
    fun loadExpensePieChart(
        context: Context?,
        chartView: com.remotearthsolutions.expensetracker.utils.ChartManagerImpl.ChartView?,
        data: List<ExpeneChartData?>?
    )
}